exercises/simple-cipher/.meta/src/reference/java/Cipher.java

```diff
@@ -32,10 +32,17 @@ public String getKey() {
     }
 
     public String encode(String plainText) {
+        String copy = plainText;
         StringBuilder ciphertext = new StringBuilder(plainText.length());
 
-        for (int index = 0; index < Math.min(plainText.length(), key.length()); index++) {
-            ciphertext.append(encodeCharacter(plainText, index));
+        while (copy.length() > 0) {
+            for (int i = 0; i < Math.min(copy.length(), key.length()); i++) {
+                ciphertext.append(encodeCharacter(copy, i));
+            }
+            copy = "";
+            for (int i = ciphertext.length(); i < plainText.length(); i++) {
+                copy = copy + plainText.charAt(i);
```

Comment:
Like this you mean?

Reply:
No, sorry if I wasn't clear. I meant why did this have to change at all? Why not keep the for loop as it used to be? Does that not work with the changes to the tests? 🙂

Reply:
With the original code, the `cipherMessageLongerThanKey` test case seems to be failing

Reply:
Oh I see 🙂 Well, I think adding the extra while loop makes it a bit confusing? Maybe it would be easier to add a small helper method which will pad or cut the key so that it's the same length as the plaintext? That is something like:
```
String sameLengthKey = getKeySameLengthAsPlaintext(plaintext, key);
for (int index = 0; index < plaintext.length(); index++) {
  ciphertext.append(encodeCharacter(plaintext, index, key));
}
```

What do you think? 🙂 It's not very important though, so feel free to keep it as it is if you like. But if so please change it back to use substring 🙂

Reply:
@FridaTveit made the relevant changes as you suggested. 👍